client/my-sites/plugins/controller.js

```diff
@@ -180,6 +217,32 @@ controller = {
 		renderPluginsBrowser( context, siteUrl );
 	},
 
+	setupPlugins: function() {
+		let selectedSite = sites.getSelectedSite();
+
+		// Not a Jetpack plan
+		if ( 0 !== selectedSite.plan.product_slug.indexOf( 'jetpack_' ) ) {
```

Comment:
I think I would move all this logic to `/plan-setup`. In plugins, we usually include all this checks & fallbacks (manage & permissions error views, etc) in the main component of the view we are going to show, so the controller only includes param processing & initialization of main components. This is what I mean: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plugins/plugin.jsx#L293  (that `/plugins/plugin.jsx` is the main component that renders a single plugin view).